zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -45,6 +46,7 @@
   // always be null in case of InterpreterSettingRef
   private String group;
   private transient Map<String, String> infos;
+  private transient Map<String, Set<String>> noteIdToParaIdsetMap;
```

Comment:
Because of every `Note` already has reference to its `Paragraphs`, i don't think it make sense maintaining another noteId - paragraphId map here, according to this variable name.

How about change this variable name to `runtimeInfosToCleared` or something similar to describe it's purpose better? That may help understanding the code better, imo.